x86/Makefile

```diff
@@ -220,12 +220,25 @@ OBJ = fastmul/bignum_emontredc_8n.o \
       secp256k1/bignum_tomont_p256k1.o \
       secp256k1/bignum_triple_p256k1.o
 
+.PHONY: bmi clean clobber
+
 %.o : %.S ; cat $< | $(PREPROCESS) | $(SPLIT) | $(MANGLE) | $(ASSEMBLE) -o $@ -
 
 libs2nbignum.a: $(OBJ) ; ar -rc libs2nbignum.a $(OBJ)
 
 clean:; rm -f libs2nbignum.a */*.o */*.correct
 
+# Dynamically regenerate or destroy the files recording which functions
+# use BMI and/or ADX instructions in the x86 versions. These won't run
+# on some older x86 machines
+
+yesbmi_functions: $(OBJ) ; for i in */*.o; do objdump -d $$i | (egrep -qi '(mulx|adcx|adox)' && grep '\.globl' `echo $$i | sed -e 's/o$$/S/'`) ; done | sed -e 's/.globl//; s/ *//g' | sort >yesbmi_functions
+nonbmi_functions: $(OBJ) ; for i in */*.o; do objdump -d $$i | (egrep -qi '(mulx|adcx|adox)' || grep '\.globl' `echo $$i | sed -e 's/o$$/S/'`) ; done | sed -e 's/.globl//; s/ *//g' | sort >nonbmi_functions
```

Comment:
If you want these to run every time rather than not re-run if files named `yesbmi_functions` or `nonbmi_functions` exist, then these must be phony targets.

Also, what is the purpose of these files (and should they be checked in)?

Reply:
Ah yes, happy to add the .PHONY target for clean and go.

The xxxbmi_functions files are just good to have as a reference for various things and it's nice to have a standard way of creating them. But they are not directly used as such and for that reason I didn't make them rebuild by default. If they actually become directly used (as with the "allowed_asm" ones) then it would certainly make sense to always rebuild them.

Reply:
Added the PHONY targets as suggested.

I also added -O3 to the compilation of the benchmarking program.